pkg/node/controller.go

```diff
@@ -39,27 +49,44 @@ func (h *handler) onRemove(key string, node *core.Node) (*core.Node, error) {
 
 func (h *handler) updateHosts(node *core.Node, removed bool) (*core.Node, error) {
 	var (
-		newHosts    string
+		nodeName    string
 		nodeAddress string
-		hostsMap    map[string]string
 	)
-	hostsMap = make(map[string]string)
-
+	nodeName = node.Name
 	for _, address := range node.Status.Addresses {
 		if address.Type == "InternalIP" {
 			nodeAddress = address.Address
 			break
 		}
 	}
-	if nodeAddress == "" {
-		logrus.Errorf("No InternalIP found for node " + node.Name)
-		return nil, nil
+	if h.modCoreDNS {
+		if err := h.updateCoreDNSConfigMap(nodeName, nodeAddress, removed); err != nil {
+			return nil, err
+		}
+	}
+	if removed {
+		if err := h.removeNodePassword(nodeName); err != nil {
+			logrus.Warn(errors.Wrap(err, "Unable to remove node password"))
+		}
+	}
+	return nil, nil
+}
+
+func (h *handler) updateCoreDNSConfigMap(nodeName, nodeAddress string, removed bool) error {
+	if nodeAddress == "" && !removed {
+		logrus.Errorf("No InternalIP found for node " + nodeName)
+		return nil
 	}
+	var (
+		newHosts string
+		hostsMap map[string]string
+	)
+	hostsMap = make(map[string]string)
```

Comment:
Can we collapse the declaration and initialization into a single line? And move the newHosts declaration down towards its use?